lib/abstractivator/trees.rb

```diff
@@ -59,6 +59,22 @@ def tree_compare(tree, mask, path=[], index=nil)
             else
               [diff(path, tree, mask)]
             end
+          when SetMask
```

Comment:
this is blatantly copied from the Enumerable case. I'd like to DRY it by building off the existing Hash support